analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/OperationProcessors/Binary.cs

```diff
@@ -72,9 +72,12 @@ private static SymbolicConstraint BinaryConstraint(BinaryOperatorKind kind, Symb
             }
             else if (left is null || right is null)
             {
-                return kind == BinaryOperatorKind.Or && (left ?? right).HasConstraint(BoolConstraint.True)
-                    ? BoolConstraint.True
-                    : null;
+                return kind switch
+                {
+                    BinaryOperatorKind.Or when (left ?? right).HasConstraint(BoolConstraint.True) => BoolConstraint.True,
+                    BinaryOperatorKind.And when (left ?? right).HasConstraint(BoolConstraint.False) => BoolConstraint.False,
```

Comment:
I don't see any reason why this isn't valid for BinaryOperatorKind.ConditionalAnd and BinaryOperatorKind.ConditionalOr respectively.

Reply:
While they are `BinaryOperatorKind` for syntax nodes, in terms of IOperation and CFG, they are not binary operations. They are encoded in the CFG shape.

But I can add them in case it was different in some early versions